lua/core/midi.lua

```diff
@@ -85,55 +85,87 @@ function Midi:send(data)
   end
 end
 
+--- send midi note on event
+-- @param integer note : note number
+-- @param integer vel : velocity 
+-- @param integer ch : midi channel
 function Midi:note_on(note, vel, ch)
   self:send{type="note_on", note=note, vel=vel, ch=ch or 1}
 end
 
+--- send midi note off event
+-- @param integer note : note number
+-- @param integer vel : velocity 
+-- @param integer ch : midi channel
 function Midi:note_off(note, vel, ch)
   self:send{type="note_off", note=note, vel=vel or 100, ch=ch or 1}
 end
 
+--- send midi continuous controller event
+-- @param integer note : cc number
+-- @param integer val : value 
+-- @param integer ch : midi channel
 function Midi:cc(cc, val, ch)
   self:send{type="cc", cc=cc, val=val, ch=ch or 1}
 end
 
+--- send midi pitchbend event
+-- @param integer val : value 
+-- @param integer ch : midi channel
 function Midi:pitchbend(val, ch)
   self:send{type="pitchbend", val=val, ch=ch or 1}
 end
 
+--- send midi key pressure event
+-- @param integer note : note number
+-- @param integer val : value 
+-- @param integer ch : midi channel
 function Midi:key_pressure(note, val, ch)
   self:send{type="key_pressure", note=note, val=val, ch=ch or 1}
 end
 
+--- send midi channel pressure event
+-- @param integer val : value 
+-- @param integer ch : midi channel
 function Midi:channel_pressure(val, ch)
   self:send{type="channel_pressure", val=val, ch=ch or 1}
 end
 
+--- send midi start event
 function Midi:start()
   self:send{type="start"}
 end
 
+--- send midi stop event
 function Midi:stop()
   self:send{type="stop"}
 end
 
+--- send midi continue event
 function Midi:continue()
   self:send{type="continue"}
 end
 
+--- send midi clock event
 function Midi:clock()
   self:send{type="clock"}
 end
 
+--- send midi song position event
+-- @param integer lsb :  
+-- @param integer msb : 
 function Midi:song_position(lsb, msb)
   self:send{type="song_position", lsb=lsb, msb=msb}
 end
 
+--- send midi song select event
+-- @param integer val : value
 function Midi:song_select(val)
   self:send{type="song_select", val=val}
 end
 
---- create device, returns object with handler and send
+--- create device, returns object with handler and send.
```

Comment:
why some function descriptions have periods and some don't?

Reply:
right.  i think they should all have a trailing period.

Reply:
Fixed periods and added data type in last commit